internal/collection/collection.go

```diff
@@ -11,3 +11,13 @@ func CopyWithSkip[T comparable](list []T, skip T) []T {
 
 	return newList
 }
+
+func MapKeysToSlice[K comparable, V any](m map[K]V) []K {
```

Comment:
This could be simply called `Keys`:
```suggestion
func Keys[K comparable, V any](m map[K]V) []K {
```